File: MagicSquare.java
import java.util.*;
class MagicSquare
{
	public static void main(String args[])
	{
		Scanner scr= new Scanner(System.in);
		int r,c,n,i,tr,tc;
		

		System.out.println("Enter the order of the Magic Square");
		n=scr.nextInt();
		if(n%2==0)
			System.out.println("Order must be a ODD number...!!1");
		
	    
	    int a[][]=new int[n][n];
	    r=0;
	    c=n/2;
	    i=1;
	    while(i<=n*n)
	    {
	    	a[r][c]=i;
	    	i++;
	    	tr=(r-1)>=0?(r-1):(n-1);
	    	tc=(c+1)%n;
	    	if(a[tr][tc]!=0)
	    		r=(r+1)%n;
	    	else 
	    	{
	    		r=tr;
	    		c=tc;

	    	}
	    }
	    System.out.println("THE MAGIC SQUARE OF ORDER : "+n);
	    for(r=0;r<n;r++)
	    {
	    	for(c=0;c<n;c++)
	    	System.out.print(a[r][c]+" ");
	    System.out.println();
	    }


	}
}

File: magicsquare.c
 #include<stdio.h>
 void magicsquareodd(int x[][10],int n)
 {
 int i,r=0,c=n/2,rtemp,ctemp;
 x[0][c]=1;
 for(i=2;i<=n*n;i++)
     {
     rtemp=r;
     ctemp=c;
     r==0?r=n-1:--r;
     c==n-1?c=0:++c;
     if(x[r][c]==0)
            x[r][c]=i;
      else {
      x[++rtemp][ctemp]=i;
      r=rtemp;
      c=ctemp;
      }  
     }
 }
 void display(int x[][10],int n)
 {
 int i,j;
 for(i=0;i<n;i++){
     for(j=0;j<n;j++)
       
       printf("%3d ",x[i][j]);
       
       printf("\n");
   }      
 }
 void main()
 {
 int x[10][10]={0},n;
 printf("Enter the order of the matrix\n");
 scanf("%d",&n);
 if(n%2){
     magicsquareodd(x,n);
     display(x,n);
     }
 else printf("order must be odd");    
 }
